pkg/deployment/deployment.go

```diff
@@ -280,8 +280,8 @@ func (d *Deployer) addCertMounts(
 				},
 			}
 			certMountDir := service.Spec.TLSCert.EDPMRoleServiceName
-			if certMountDir == "" && service.Spec.EDPMServiceName != "" {
-				certMountDir = service.Spec.EDPMServiceName
+			if certMountDir == "" && service.Spec.EDPMServiceType != "" {
+				certMountDir = service.Spec.EDPMServiceType
 			} else {
```

Comment:
shouldn't it be else if used here? as even with EDPMRoleServiceName set else condition would get executed.

Reply:
it depends.
yes if we intend to keep EDPMRoleServiceName but I'm not sure that is the intent.

at least I'm not convinced we should but it would not be incorrect to make this an elif for now

Reply:
yes i was just referring current scenario on how these used currently as it doesn't match what described in the comments.

Reply:
i fixed the if/else in the webhook change i made.

For now, I have kept EDPMRoleServiceName, I think we can drop that in a future PR if we want. But for now, it's the only way to override the mount dir for a specific cert (if needed).